contracts/secret-storage/SecretManager.sol

```diff
@@ -601,42 +608,31 @@ contract SecretManager is
         // if(!userStorage[_secretId].selectedEnclaves[enclaveIndex].hasAcknowledgedStore || userStorage[_secretId].ackTimestamp == 0)
         //     return;
 
-        _updateUsdcDepositPostPayment(_secretId, enclaveIndex, _lastAliveTimestamp);
+        uint256 ackTimestamp = _getSecretStoreAckTimestamp(_secretId, enclaveIndex);
+        uint256 endTimestamp = userStorage[_secretId].endTimestamp;
+        _updateUsdcDepositPostPayment(_secretId, ackTimestamp, endTimestamp, _lastAliveTimestamp);
 
         _replaceStore(_secretId, _enclaveAddress, enclaveIndex, true);
     }
 
     function _renounceSecrets(
         address _enclaveAddress,
-        address _owner
-    ) internal {
-        ( , address owner, , bool draining) = TEE_MANAGER.teeNodes(_enclaveAddress);
-        if(!draining)
-            revert SecretManagerStoreNotDraining();
-        // can use signature verification as well
-        if(_owner != owner)
-            revert SecretManagerInvalidSecretOwner();
-
-        uint256 lastAliveTimestamp = SECRET_STORE.getSecretStoreLastAliveTimestamp(_enclaveAddress);
-        SECRET_STORE.renounceSecretsPreUpdate(_enclaveAddress, MARK_ALIVE_TIMEOUT, STAKING_PAYMENT_POOL);
-
-        // TODO: how to handle the newly assigned unacknowledged secrets
-        uint256[] memory storeAckSecretIds = SECRET_STORE.getStoreAckSecretIds(_enclaveAddress);
-        uint256 len = storeAckSecretIds.length;
-        uint256 occupiedStorage;
+        address _owner,
+        uint256[] memory _storeAckSecretIds,
+        uint256 _lastAliveTimestamp
+    ) internal returns (uint256 occupiedStorage) {
+        uint256 len = _storeAckSecretIds.length;
         uint256 storageTimeUsage;
         for (uint256 index = 0; index < len; index++) {
-            uint256 secId = storeAckSecretIds[index];
+            uint256 secId = _storeAckSecretIds[index];
             uint256 sizeLimit = userStorage[secId].sizeLimit;
             occupiedStorage += userStorage[secId].sizeLimit;
```

Comment:
remove this, not required? Is it compiling?

Reply:
yeah, not a syntax error